x/blob/types/payforblob.go

```diff
@@ -61,10 +61,10 @@ func (msg *MsgPayForBlob) Type() string {
 // ValidateBasic fulfills the sdk.Msg interface by performing stateless
 // validity checks on the msg that also don't require having the actual blob
 func (msg *MsgPayForBlob) ValidateBasic() error {
-	if len(msg.NamespaceIds) != len(msg.ShareVersions) || len(msg.NamespaceIds) != len(msg.BlobSizes) {
+	if len(msg.NamespaceIds) != len(msg.ShareVersions) || len(msg.NamespaceIds) != len(msg.BlobSizes) || len(msg.NamespaceIds) != len(msg.ShareCommitments) {
```

Comment:
Do we also want to add a check that this length is not zero?

Reply:
By this length are you referring to msg.ShareCommitments or msg.NamespaceIds? So something like this:

```go
	if len(msg.NamespaceIds) == 0 {
		return ErrNoNamespaceIds
	}
```

or 

```go
	if len(msg.ShareCommitments) == 0 {
		return ErrNoShareCommitments
	}
```

Reply:
IMO both would work. We could include checks that all of these fields are non-zero: NamespaceIds, ShareVersions, BlobSizes, and ShareCommitments. Will do in a follow-up PR.